test/object-store/object.cpp

```diff
@@ -354,29 +385,431 @@ TEST_CASE("object") {
+            auto col_target_backlink = table_origin->get_opposite_column(col_origin_link);
+
+            r->begin_transaction();
+
+            Obj obj_target = table_target->create_object_with_primary_key(200);
+            Object object_target(r, obj_target);
+            object_target.set_column_value("value 1", 201);
+            object_target.set_column_value("value 2", 202);
+
+            Obj obj_origin = table_origin->create_object_with_primary_key(100);
+            Object object_origin(r, obj_origin);
+            object_origin.set_column_value("value", 101);
+            object_origin.set_property_value(d, "link", util::Any(object_target));
+
+            r->commit_transaction();
+
+            std::pair<TableKey, ColKey> pair_origin_value(table_origin->get_key(), col_origin_value);
+            std::pair<TableKey, ColKey> pair_origin_link(table_origin->get_key(), col_origin_link);
+            std::pair<TableKey, ColKey> pair_origin_link2(table_origin->get_key(), col_origin_link2);
+            std::pair<TableKey, ColKey> pair_target_backlink1(table_target->get_key(), col_target_backlink);
+            std::pair<TableKey, ColKey> pair_target_value1(table_target->get_key(), col_target_value1);
+            std::pair<TableKey, ColKey> pair_target_value2(table_target->get_key(), col_target_value2);
+
+            auto key_path_origin_value = {pair_origin_value};
+            auto key_path_origin_link = {pair_origin_link};
+            auto key_path_target_value1 = {pair_target_value1};
+            auto key_path_target_value2 = {pair_target_value2};
+
+            auto key_path_origin_to_target_value1 = {pair_origin_link, pair_target_value1};
+            auto key_path_origin_to_target_value2 = {pair_origin_link, pair_target_value2};
+            auto key_path_target_backlink = {pair_target_backlink1};
+            auto key_path_target_to_origin_value = {pair_target_backlink1, pair_origin_value};
+
+            KeyPathArray key_path_array_origin_value = {key_path_origin_value};
+            KeyPathArray key_path_array_origin_link = {key_path_origin_link};
+            KeyPathArray key_path_array_target_value1 = {key_path_target_value1};
+            KeyPathArray key_path_array_target_value2 = {key_path_target_value2};
+
+            KeyPathArray key_path_array_origin_to_target_value1 = {key_path_origin_to_target_value1};
+            KeyPathArray key_path_array_origin_to_target_value2 = {key_path_origin_to_target_value2};
+            KeyPathArray key_path_array_target_backlink = {key_path_target_backlink};
+            KeyPathArray key_path_array_target_to_origin_value = {key_path_target_to_origin_value};
+
+            SECTION("callbacks on a single object") {
+                SECTION("modifying origin table 'table2', property 'value'"
+                        "while observing origin table 'table2', property 'value'"
+                        "-> DOES send a notification") {
```

Comment:
One thing I notice from the failing tests is that these test names all need some leading or trailing spaces inside the quotes when they're line-wrapped.

Reply:
@ericjordanmossman @jsflax The 8 failing tests are the 2x2x2 additional tests I mentioned in the Slack thread that we need to check if that's expected or not.
Let me know your final thoughts on that.
We could also just merge this one for now (since it's pretty big and I already had merge conflicts because master was changes) and then add another PR if we see additional need.

I would of course remove the failing test cases first and make sure that the PR is green (also re-merge master).

Reply:
I would imagine they would be included, because I believe these tests are the equivalent of the list behavior but with backlinks?

Reply:
I would not describe it that way but maybe that's correct too.
They are tests that check if an insertion in a forward link leads to a modification in a backlink.
Why they don't show up in the transaction change set at all (that's the part that already existed - untouched by me) puzzles me though. Because I can't filter what doesn't exist. If we need further changes here I'd recommend adding them with a new PR.

Reply:
The source tables for incoming links needs to be added to the thing populated by find_related_tables(). Part of the KPF changes is adding some notifications where ones previously didn't exist, specifically when linking objects properties are included in the filter.